### DISCUSSION TITLE
NLP ingredient parsing not working at all

### DISCUSSION THREAD

**Discussion by rcdailey** [mealie-recipes/mealie#2546] [Q&A] [answered]
I must have missed a basic memo somewhere, but I can't get NLP ingredient parsing working at all. In my brand new instance of Mealie, I created a new recipe with nothing set up except a name. I disabled "Disable Ingredient Amounts".

I then went to "Bulk Add" for ingredients and typed in the following:

```txt
1 cup water
1/2 cup sugar
```

The end result looks like this:

![image](https://github.com/mealie-recipes/mealie/assets/1768054/109f8be9-4459-4a26-b695-82a9b1376c99)

I then clicked "Parse".

First annoyance here is the browser alert that asks me to save before leaving the page (this prompt should disappear in favor of the parse UI being a modal over the recipe view so a user isn't required to save changes):

![image](https://github.com/mealie-recipes/mealie/assets/1768054/67956563-45fb-4f82-8dfd-448e49eed306)

This next page is where I am completely lost:

![image](https://github.com/mealie-recipes/mealie/assets/1768054/a64c1953-90ed-43f5-b8f3-dabcbd2b4df1)

1. Instructions at the top aren't really helpful. I click the "Parse All" button, but nothing happens.
2. How do I know for sure that the "notes" section is where I should manually type ingredients?
3. Why didn't Mealie auto fill in the quantity, unit, and food fields?

What I was expecting was Mealie to be smart enough to know that "1" is the quantity, "cup" is the unit, and "sugar" is the food.

It's unclear to me if that's the intended behavior or if I'm supposed to manually type all of these into the individual fields first and only when they already exist does Mealie match (and auto select) them. If so, that seems to defeat the purpose IMO, as it'll be a lot of up-front work to type in these ingredients that way.

Are my expectations the issue here?

**Comment by michael-genson** [comment, marked as answer]
The intent is for you to parse the ingredients into your food and unit stores. If you have no foods or units defined, you're not going to get any results


You can add foods/units from your data management page (click on your name in the upper left, then click "Manage Data"). Most people opt to seed common foods/units, which you can do from that page as well

That's what the "create missing food" button is there for. The parser determined "water" was in that ingredient, but is missing from your food store. It's supposed to do that for units too, but that's broken, [I have a PR open to fix that](https://github.com/mealie-recipes/mealie/pull/2535), among other things

**Reply by rcdailey** [reply]
Thank you for the clarification.
